chalice/deploy/planner.py

```diff
@@ -125,23 +138,30 @@ def __init__(self, remote_state, osutils):
         self._osutils = osutils
 
     def execute(self, resources):
-        # type: (List[models.Model]) -> List[models.Instruction]
+        # type: (List[models.Model]) -> models.Plan
         plan = []  # type: List[models.Instruction]
+        messages = {}  # type: Dict[int, str]
         for resource in resources:
-            name = 'plan_%s' % resource.__class__.__name__.lower()
+            name = '_plan_%s' % resource.__class__.__name__.lower()
             handler = getattr(self, name, None)
             if handler is not None:
                 result = handler(resource)
                 if result:
-                    plan.extend(result)
-        return plan
+                    for single in result:
```

Comment:
This is getting pretty nested. I wonder if it makes sense to add a helper method to clean this up a bit? Maybe something like`_add_result(self, result, plan, messages)`?

Reply:
Updated.